crypto/pem/pem_local.h

```diff
@@ -98,16 +98,16 @@
     return PEM_ASN1_##writename##((i2d_of_void *)i2d_##asn1, str, out,  \
                                   x, enc, kstr, klen, cb, u)
 
-# define IMPLEMENT_PEM_provided_write_to(name, type, str, asn1,         \
+# define IMPLEMENT_PEM_provided_write_to(name, TYPE, type, str, asn1,   \
                                          OUTTYPE, outtype, writename)   \
-    PEM_write_fnsig(name, type, OUTTYPE, writename)                     \
+    PEM_write_fnsig(name, TYPE, OUTTYPE, writename)                     \
     {                                                                   \
         IMPLEMENT_PEM_provided_write_body_vars(type, asn1, NULL);       \
         IMPLEMENT_PEM_provided_write_body_main(type, outtype);          \
```

Comment:
This looks weird.  You have uppercased `type` to `TYPE` in some places, but not others.  Quite frankly, I don't see a reason for uppercasing that one at all, and would suggest simply undoing it.  It's a distraction for reviewing.

Reply:
Please note that we need both type and TYPE because the TYPE is EVP_PKEY and type is just pkey here. Maybe the lowercase type should be renamed somehow, I welcome suggestions on how.

Reply:
Ahhhhhh, I totally missed that point.  Ok, sure, this is fine with that in mind.